components/styled/table/src/molecules/table.tsx

```diff
@@ -1,192 +1,46 @@
-    if (cellIdx === 0) {
-      return 0
-    }
-
-    if (cellIdx < staticColumns) {
-      return firstColumnWidth
-    }
-
-    return undefined
-  }
-
-  return (
-    <Container>
-      <StyledTable>
-        <thead>
-          {table.getHeaderGroups().map((headerGroup) => (
-            <tr key={headerGroup.id}>
-              {headerGroup.headers.map((header, headerIndex) =>
-                headerIndex === 0 ? (
-                  <StyledTableHeader
-                    stickyWidth={calculateStaticColumnOffset(headerIndex, staticColumns, width)}
-                    key={header.id}
-                    colSpan={header.colSpan}
-                    ref={firstColumn}
-                  >
-                    {header.isPlaceholder ? null : flexRender(header.column.columnDef.header, header.getContext())}
-                  </StyledTableHeader>
-                ) : (
-                  <StyledTableHeader
-                    stickyWidth={calculateStaticColumnOffset(headerIndex, staticColumns, width)}
-                    key={header.id}
-                    colSpan={header.colSpan}
-                    {...{
-                      style: {
-                        cursor: header.column.getCanSort() ? 'pointer' : '',
-                      },
-                      onClick: header.column.getToggleSortingHandler(),
-                    }}
-                  >
-                    {header.isPlaceholder ? null : flexRender(header.column.columnDef.header, header.getContext())}
-                  </StyledTableHeader>
-                )
-              )}
-            </tr>
-          ))}
-        </thead>
-        <tbody>
-          {table.getRowModel().rows.map((row) => (
-            <tr key={row.id}>
-              {row.getVisibleCells().map((cell, cellIdx) => (
-                <StyledTableData
-                  stickyWidth={calculateStaticColumnOffset(cellIdx, staticColumns, width)}
-                  key={cell.id}
-                  style={{
-                    background: cellIdx % 2 === 1 ? TABLE_COLOURS.STRIPE_LIGHT : TABLE_COLOURS.STRIPE_DARK,
-                    textAlign: 'center',
-                  }}
-                >
-                  {flexRender(cell.column.columnDef.cell, cell.getContext())}
-                </StyledTableData>
-              ))}
-            </tr>
-          ))}
-        </tbody>
-      </StyledTable>
-    </Container>
+import { FC } from 'react'
+import { Axis } from '@ltht-react/types'
+import InfiniteScrollManualPaginatedTable from './infinite-scroll-manual-paginated-table'
+import InfiniteScrollAutoPaginatedTable from './infinite-scroll-auto-paginated-table'
+import { TableData } from './table-core'
+
+const Table: FC<IProps> = ({
+  tableData,
+  staticColumns = 0,
+  currentPage = 1,
+  pageSize = 10,
+  headerAxis = 'x',
+  manualPagination = false,
+  getCanNextPage = () => false,
+  nextPage = () => null,
+}) =>
+  manualPagination ? (
```

Comment:
I'd still much rather keep this as a single component rather than creating 3 separate ones. Looking at the differences, there's a lot of shared code (they both return identical JSX blocks and they both invoke `useReactTable` in the same way bar one option which can be computed). 
The only major difference seems to be the content of the useEffect hook which I think we could abstract